src/NServiceBus.Core/Transports/UnicastTransportOperation.cs

```diff
@@ -0,0 +1,44 @@
+namespace NServiceBus.Transports
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using NServiceBus.DeliveryConstraints;
+    using NServiceBus.Routing;
+
+    /// <summary>
+    /// Represents a transport operation which should be delivered to a single receiver.
+    /// </summary>
+    public class UnicastTransportOperation
+    {
+        /// <summary>
+        /// Creates a new <see cref="UnicastTransportOperation"/> instance.
+        /// </summary>
+        public UnicastTransportOperation(OutgoingMessage message, UnicastAddressTag addressTag, IEnumerable<DeliveryConstraint> deliveryConstraints = null, DispatchConsistency requiredDispatchConsistency = DispatchConsistency.Default)
+        {
+            Message = message;
+            AddressTag = addressTag;
+            DeliveryConstraints = deliveryConstraints ?? Enumerable.Empty<DeliveryConstraint>();
+            RequiredDispatchConsistency = requiredDispatchConsistency;
+        }
+
+        /// <summary>
+        /// The message to be sent over the transport.
+        /// </summary>
+        public OutgoingMessage Message { get; set; }
+
+        /// <summary>
+        /// Defines the receiver of the message.
+        /// </summary>
+        public UnicastAddressTag AddressTag { get; set; }
```

Comment:
Agree with @johnsimons here. While there may be some value in these wrapper classes internally i just see friction forcing them onto users. Considering many of our own team, when seeing them, respond with "huh why, what is the value of this" i think the consumers of our api will have problems groc'ing the concept.